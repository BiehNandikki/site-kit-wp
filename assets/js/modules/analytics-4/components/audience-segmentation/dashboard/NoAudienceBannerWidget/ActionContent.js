/**
 * ActionContent component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { createInterpolateElement } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import Link from '../../../../../../components/Link';
import { CORE_LOCATION } from '../../../../../../googlesitekit/datastore/location/constants';
import { CORE_SITE } from '../../../../../../googlesitekit/datastore/site/constants';
import useViewOnly from '../../../../../../hooks/useViewOnly';

export default function ActionContent( { hasConfigurableAudiences } ) {
	const isViewOnly = useViewOnly();

	const settingsURL = useSelect( ( select ) =>
		select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' )
	);

	const { navigateTo } = useDispatch( CORE_LOCATION );

	function hideAudiencesWidget() {
		// TODO: dismiss item...
	}

	if ( isViewOnly ) {
		if ( ! hasConfigurableAudiences ) {
			return (
				<p>
					{ createInterpolateElement(
						__(
							'You can <a>temporarily hide</a> this feature until groups are available.',
							'google-site-kit'
						),
						{
							a: (
								<Link
									secondary
									onClick={ hideAudiencesWidget }
								/>
							),
						}
					) }
				</p>
			);
		}

		return null;
	}

	return (
		<p>
			{ createInterpolateElement(
				__(
					'You can deactivate this widget in <a>Settings</a>.',
					'google-site-kit'
				),
				{
					a: (
						<Link
							secondary
							onClick={ () =>
								navigateTo( `${ settingsURL }#/admin-settings` )
							}
						/>
					),
				}
			) }
		</p>
	);
}

ActionContent.propTypes = {
	hasConfigurableAudiences: PropTypes.bool.isRequired,
};
