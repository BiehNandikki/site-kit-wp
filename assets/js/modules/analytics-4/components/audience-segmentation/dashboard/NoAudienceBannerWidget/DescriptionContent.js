/**
 * DescriptionContent component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { createInterpolateElement } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import Link from '../../../../../../components/Link';
import { CORE_SITE } from '../../../../../../googlesitekit/datastore/site/constants';
import { CORE_UI } from '../../../../../../googlesitekit/datastore/ui/constants';
import { CORE_USER } from '../../../../../../googlesitekit/datastore/user/constants';
import { AUDIENCE_SELECTION_PANEL_OPENED_KEY } from '../AudienceSelectionPanel/constants';
import useViewOnly from '../../../../../../hooks/useViewOnly';

export default function DescriptionContent( { hasConfigurableAudiences } ) {
	const isViewOnly = useViewOnly();

	const hasConfiguredAudiences = useSelect(
		( select ) => select( CORE_USER ).getConfiguredAudiences().length > 0
	);

	const didSetAudiences = useSelect( ( select ) =>
		select( CORE_USER ).didSetAudiences()
	);

	const audienceLearnMoreURL = useSelect( ( select ) =>
		select( CORE_SITE ).getGoogleSupportURL( {
			path: '/analytics/answer/12799087',
		} )
	);

	const { setValue } = useDispatch( CORE_UI );

	if ( ! hasConfigurableAudiences ) {
		if ( ! hasConfiguredAudiences && ( isViewOnly || ! didSetAudiences ) ) {
			return (
				<p>
					{ createInterpolateElement(
						__(
							'There are no visitor groups available. Learn more about how to group site visitors in <a>Analytics</a>.',
							'google-site-kit'
						),
						{
							a: (
								<Link
									secondary
									href={ audienceLearnMoreURL }
									external
									hideExternalIndicator
								/>
							),
						}
					) }
				</p>
			);
		}

		return (
			<p>
				{ createInterpolateElement(
					__(
						'It looks like your visitor groups aren’t available anymore. Learn more about how to group site visitors in <a>Analytics</a>.',
						'google-site-kit'
					),
					{
						a: (
							<Link
								secondary
								href={ audienceLearnMoreURL }
								external
								hideExternalIndicator
							/>
						),
					}
				) }
			</p>
		);
	}

	return (
		<p>
			{ createInterpolateElement(
				__(
					'It looks like your visitor groups aren’t available anymore. <a>Select other groups</a>.',
					'google-site-kit'
				),
				{
					a: (
						<Link
							secondary
							onClick={ () =>
								setValue(
									AUDIENCE_SELECTION_PANEL_OPENED_KEY,
									true
								)
							}
						/>
					),
				}
			) }
		</p>
	);
}

DescriptionContent.propTypes = {
	hasConfigurableAudiences: PropTypes.bool.isRequired,
};
