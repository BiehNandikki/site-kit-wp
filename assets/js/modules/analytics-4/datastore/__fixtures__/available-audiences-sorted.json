[
  {
    "audienceSlug": "",
    "audienceType": "USER_AUDIENCE",
    "description": "Description",
    "displayName": "Test Audience",
    "name": "properties/12345/audiences/5"
  },
  {
    "audienceSlug": "new-visitors",
    "audienceType": "SITE_KIT_AUDIENCE",
    "description": "People who visited the site for the first time",
    "displayName": "New visitors",
    "name": "properties/12345/audiences/3"
  },
  {
    "adsPersonalizationEnabled": true,
    "audienceSlug": "returning-visitors",
    "audienceType": "SITE_KIT_AUDIENCE",
    "description": "People who have visited your site at least once before",
    "displayName": "Returning visitors",
    "name": "properties/12345/audiences/4"
  },
  {
    "audienceSlug": "all-users",
    "audienceType": "DEFAULT_AUDIENCE",
    "description": "All users",
    "displayName": "All visitors",
    "name": "properties/12345/audiences/1"
  },
  {
    "audienceSlug": "purchasers",
    "audienceType": "DEFAULT_AUDIENCE",
    "description": "Users who have made a purchase",
    "displayName": "Purchasers",
    "name": "properties/12345/audiences/2"
  }
]
