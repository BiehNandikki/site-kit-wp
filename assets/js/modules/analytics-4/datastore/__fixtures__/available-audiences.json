[
    {
        "name": "properties/437749618/audiences/8004992724",
        "displayName": "All visitors",
        "description": "All users",
        "audienceType": "DEFAULT_AUDIENCE",
        "audienceSlug": "all-users"
    },
    {
        "name": "properties/437749618/audiences/8004992725",
        "displayName": "Purchasers",
        "description": "Users who have made a purchase",
        "audienceType": "DEFAULT_AUDIENCE",
        "audienceSlug": "purchasers"
    },
    {
        "name": "properties/437749618/audiences/8030142001",
        "displayName": "Test Audience",
        "description": "This is the test description for test audience.",
        "audienceType": "SITE_KIT_AUDIENCE",
        "audienceSlug": "returning-visitors"
    }
]
