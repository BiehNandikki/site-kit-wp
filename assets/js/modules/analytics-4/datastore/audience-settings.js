/**
 * `modules/analytics-4` data store: audience settings.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
// import invariant from 'invariant';
// import { isPlainObject } from 'lodash';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import Data from 'googlesitekit-data';
import { MODULES_ANALYTICS_4 } from './constants';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
// import { createValidatedAction } from '../../../googlesitekit/data/utils';
import { createReducer } from '../../../googlesitekit/data/create-reducer';
// import { actions as errorStoreActions } from '../../../googlesitekit/data/create-error-store';

// const { receiveError, clearError } = errorStoreActions;

const fetchStoreReducerCallback = createReducer(
	( state, audienceSettings ) => {
		if ( ! state.audienceSettings ) {
			state.audienceSettings = {};
		}

		state.audienceSettings.settings = audienceSettings;
		state.audienceSettings.savedSettings = audienceSettings;
	}
);

const fetchGetAudienceSettingsStore = createFetchStore( {
	baseName: 'getAudienceSettings',
	controlCallback() {
		return API.get(
			'modules',
			'analytics-4',
			'audience-settings',
			{},
			{
				useCache: false,
			}
		);
	},
	reducerCallback: fetchStoreReducerCallback,
} );

const baseInitialState = {
	audienceSettings: undefined,
};

const baseActions = {};

const baseControls = {};

const baseReducer = ( state, { type } ) => {
	switch ( type ) {
		default: {
			return state;
		}
	}
};

const baseResolvers = {
	*getAudienceSettings() {
		const registry = yield Data.commonActions.getRegistry();

		const audienceSettings = registry
			.select( MODULES_ANALYTICS_4 )
			.getAudienceSettings();

		if ( audienceSettings === undefined ) {
			yield fetchGetAudienceSettingsStore.actions.fetchGetAudienceSettings();
		}
	},
};

const baseSelectors = {
	/**
	 * Gets the audience settings.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object} state Data store's state.
	 * @return {(Object|undefined)} Audience settings; `undefined` if not loaded.
	 */
	getAudienceSettings( state ) {
		return state.audienceSettings;
	},
};

const store = Data.combineStores( fetchGetAudienceSettingsStore, {
	initialState: baseInitialState,
	actions: baseActions,
	controls: baseControls,
	reducer: baseReducer,
	resolvers: baseResolvers,
	selectors: baseSelectors,
} );

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export default store;
