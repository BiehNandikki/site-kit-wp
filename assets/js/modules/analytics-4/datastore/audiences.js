/**
 * `modules/analytics-4` data store: audiences.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';

const fetchGetAudiencesStore = createFetchStore( {
	baseName: 'getAudiences',
	controlCallback() {
		return API.get(
			'modules',
			'analytics-4',
			'audiences',
			{},
			{
				useCache: false,
			}
		);
	},
	reducerCallback( state, audiences ) {
		return { ...state, audiences };
	},
} );
fetchGetAudiencesStore();
