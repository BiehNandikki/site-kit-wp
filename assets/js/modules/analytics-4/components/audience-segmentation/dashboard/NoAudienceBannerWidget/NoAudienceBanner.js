/**
 * NoAudienceBanner component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { createInterpolateElement } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import NoAudienceBannerGraphic from '../../../../../../../svg/graphics/no-audience-banner-graphic.svg';
import Link from '../../../../../../components/Link';
import { CORE_MODULES } from '../../../../../../googlesitekit/modules/datastore/constants';
import LeanCTABanner from '../../../../../../components/LeanCTABanner';
import { CORE_LOCATION } from '../../../../../../googlesitekit/datastore/location/constants';
import { CORE_SITE } from '../../../../../../googlesitekit/datastore/site/constants';
import { CORE_UI } from '../../../../../../googlesitekit/datastore/ui/constants';
import { AUDIENCE_SELECTION_PANEL_OPENED_KEY } from '../AudienceSelectionPanel/constants';

export default function NoAudienceBanner( { hasConfigurableAudiences } ) {
	const Icon = useSelect( ( select ) =>
		select( CORE_MODULES ).getModuleIcon( 'analytics-4' )
	);
	const settingsURL = useSelect( ( select ) =>
		select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' )
	);
	const audienceLearnMoreURL = useSelect( ( select ) =>
		select( CORE_SITE ).getGoogleSupportURL( {
			path: '/analytics/answer/12799087',
		} )
	);

	const { setValue } = useDispatch( CORE_UI );
	const { navigateTo } = useDispatch( CORE_LOCATION );

	return (
		<LeanCTABanner
			className="googlesitekit-no-audience-banner"
			Icon={ Icon }
			SVGGraphic={ NoAudienceBannerGraphic }
		>
			<p className="googlesitekit-lean-cta-banner--body__text">
				{ hasConfigurableAudiences &&
					createInterpolateElement(
						__(
							'It looks like your visitor groups aren’t available anymore. <a>Select other groups</a>.',
							'google-site-kit'
						),
						{
							a: (
								<Link
									secondary
									onClick={ () =>
										setValue(
											AUDIENCE_SELECTION_PANEL_OPENED_KEY,
											true
										)
									}
								/>
							),
						}
					) }
				{ ! hasConfigurableAudiences &&
					createInterpolateElement(
						__(
							'It looks like your visitor groups aren’t available anymore. Learn more about how to group site visitors in <a>Analytics</a>.',
							'google-site-kit'
						),
						{
							a: (
								<Link
									secondary
									href={ audienceLearnMoreURL }
									external
									hideExternalIndicator
								/>
							),
						}
					) }
			</p>
			<p className="googlesitekit-lean-cta-banner--body__text">
				{ createInterpolateElement(
					__(
						'You can deactivate this widget in <a>Settings</a>.',
						'google-site-kit'
					),
					{
						a: (
							<Link
								secondary
								onClick={ () =>
									navigateTo(
										`${ settingsURL }#/admin-settings`
									)
								}
							/>
						),
					}
				) }
			</p>
		</LeanCTABanner>
	);
}

NoAudienceBanner.propTypes = {
	hasConfigurableAudiences: PropTypes.bool.isRequired,
};
