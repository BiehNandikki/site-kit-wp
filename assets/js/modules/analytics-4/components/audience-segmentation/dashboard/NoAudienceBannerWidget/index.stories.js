/**
 * NoAudienceBannerWidget Component Stories.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies.
 */
import NoAudienceBannerWidget from '.';
import WithRegistrySetup from '../../../../../../../../tests/js/WithRegistrySetup';
import {
	provideModuleRegistrations,
	provideModules,
} from '../../../../../../../../tests/js/utils';
import { withWidgetComponentProps } from '../../../../../../googlesitekit/widgets/util';
import { MODULES_ANALYTICS_4 } from '../../../../datastore/constants';
import { availableAudiences } from '../../../../datastore/__fixtures__';

const WidgetWithComponentProps = withWidgetComponentProps(
	'analyticsNoAudienceBanner'
)( NoAudienceBannerWidget );

function Template() {
	return <WidgetWithComponentProps />;
}

export const HasConfigurableAudiences = Template.bind( {} );
HasConfigurableAudiences.storyName = 'Has Configurable Audiences';
HasConfigurableAudiences.scenario = {
	label: 'Modules/Analytics4/Components/AudienceSegmentation/Dashboard/NoAudienceBannerWidget/HasConfigurableAudiences',
};
HasConfigurableAudiences.args = {
	setupRegistry: ( registry ) => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );
	},
};

export const NoConfigurableAudience = Template.bind( {} );
NoConfigurableAudience.storyName = 'No Configurable Audience';
NoConfigurableAudience.scenario = {
	label: 'Modules/Analytics4/Components/AudienceSegmentation/Dashboard/NoAudienceBannerWidget/NoConfigurableAudience',
};
NoConfigurableAudience.args = {
	setupRegistry: ( registry ) => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setAvailableAudiences( [] );
	},
};

export default {
	title: 'Modules/Analytics4/Components/AudienceSegmentation/Dashboard/NoAudienceBannerWidget',
	decorators: [
		( Story, { args } ) => {
			const setupRegistry = async ( registry ) => {
				provideModules( registry, [
					{
						active: true,
						connected: true,
						slug: 'analytics-4',
					},
				] );
				provideModuleRegistrations( registry );

				const audienceSettings = {
					configuredAudiences: [ 'properties/12345/audiences/1' ],
					isAudienceSegmentationWidgetHidden: false,
				};

				registry
					.dispatch( MODULES_ANALYTICS_4 )
					.receiveGetAudienceSettings( audienceSettings );

				await args?.setupRegistry( registry );
			};
			return (
				<WithRegistrySetup func={ setupRegistry }>
					<Story />
				</WithRegistrySetup>
			);
		},
	],
};
