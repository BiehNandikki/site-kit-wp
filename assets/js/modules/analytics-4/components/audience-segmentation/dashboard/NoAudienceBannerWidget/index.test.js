/**
 * NoAudienceBannerWidget component tests.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import NoAudienceBannerWidget from '.';
import { render } from '../../../../../../../../tests/js/test-utils';
import {
	createTestRegistry,
	muteFetch,
	provideModuleRegistrations,
	provideModules,
	unsubscribeFromAll,
} from '../../../../../../../../tests/js/utils';
import { getWidgetComponentProps } from '../../../../../../googlesitekit/widgets/util';
import { MODULES_ANALYTICS_4 } from '../../../../datastore/constants';
import { availableAudiences } from '../../../../datastore/__fixtures__';

describe( 'NoAudienceBannerWidget', () => {
	let registry;

	const { Widget, WidgetNull } = getWidgetComponentProps(
		'analyticsNoAudienceBanner'
	);

	const auduenceSettingsRegExp = new RegExp(
		'^/google-site-kit/v1/modules/analytics-4/data/audience-settings'
	);

	beforeEach( () => {
		registry = createTestRegistry();
		provideModules( registry, [
			{
				active: true,
				connected: true,
				slug: 'analytics-4',
			},
		] );
		provideModuleRegistrations( registry );
		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {} );
	} );

	afterEach( () => {
		unsubscribeFromAll( registry );
		jest.clearAllMocks();
	} );

	it( 'should not render when availableAudiences and configuredAudiences are not loaded', () => {
		muteFetch( auduenceSettingsRegExp );

		const { container } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when configuredAudiences is not loaded', () => {
		muteFetch( auduenceSettingsRegExp );

		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		const { container } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when configured audience is matching available audiences', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/1' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when all configured audiences are matching available audiences', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [
				'properties/12345/audiences/1',
				'properties/12345/audiences/3',
			],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when some configured audiences are matching available audiences', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [
				'properties/12345/audiences/1',
				'properties/12345/audiences/9',
			],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should render with correct message when there are additional configurable audiences available', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/9' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container, getByText } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect(
			container.querySelector( '.googlesitekit-no-audience-banner' )
		).toBeInTheDocument();

		expect( getByText( /Select other groups/i ) ).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );

	it( 'should render with correct message when there is no additional configurable audience available', () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setAvailableAudiences( [] );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/9' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container, getByText } = render(
			<NoAudienceBannerWidget
				Widget={ Widget }
				WidgetNull={ WidgetNull }
			/>,
			{
				registry,
			}
		);

		expect(
			container.querySelector( '.googlesitekit-no-audience-banner' )
		).toBeInTheDocument();

		expect(
			getByText( /Learn more about how to group site visitors in/i )
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );
} );
