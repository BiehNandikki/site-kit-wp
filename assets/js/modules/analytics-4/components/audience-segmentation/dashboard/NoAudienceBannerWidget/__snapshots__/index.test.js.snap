// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoAudienceBannerWidget should render when there is no configured audience. 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsNoAudienceBanner googlesitekit-widget--no-padding"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="googlesitekit-lean-cta-banner googlesitekit-no-audience-banner"
      >
        <div
          class="googlesitekit-lean-cta-banner__body"
        >
          <div
            class="googlesitekit-lean-cta-banner__body-icon"
          >
            <svg />
          </div>
          <div
            class="googlesitekit-lean-cta-banner__body-content"
          >
            <p>
              It looks like your visitor groups aren’t available anymore. 
              <button
                class="googlesitekit-cta-link googlesitekit-cta-link--secondary"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Select other groups
                </span>
              </button>
              .
            </p>
            <p>
              You can deactivate this widget in 
              <button
                class="googlesitekit-cta-link googlesitekit-cta-link--secondary"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Settings
                </span>
              </button>
              .
            </p>
          </div>
        </div>
        <div
          class="googlesitekit-lean-cta-banner__graphic"
        >
          <svg />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NoAudienceBannerWidget should render with correct message when there are additional configurable audiences available 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsNoAudienceBanner googlesitekit-widget--no-padding"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="googlesitekit-lean-cta-banner googlesitekit-no-audience-banner"
      >
        <div
          class="googlesitekit-lean-cta-banner__body"
        >
          <div
            class="googlesitekit-lean-cta-banner__body-icon"
          >
            <svg />
          </div>
          <div
            class="googlesitekit-lean-cta-banner__body-content"
          >
            <p>
              It looks like your visitor groups aren’t available anymore. 
              <button
                class="googlesitekit-cta-link googlesitekit-cta-link--secondary"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Select other groups
                </span>
              </button>
              .
            </p>
            <p>
              You can deactivate this widget in 
              <button
                class="googlesitekit-cta-link googlesitekit-cta-link--secondary"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Settings
                </span>
              </button>
              .
            </p>
          </div>
        </div>
        <div
          class="googlesitekit-lean-cta-banner__graphic"
        >
          <svg />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NoAudienceBannerWidget should render with correct message when there is no additional configurable audience available 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsNoAudienceBanner googlesitekit-widget--no-padding"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="googlesitekit-lean-cta-banner googlesitekit-no-audience-banner"
      >
        <div
          class="googlesitekit-lean-cta-banner__body"
        >
          <div
            class="googlesitekit-lean-cta-banner__body-icon"
          >
            <svg />
          </div>
          <div
            class="googlesitekit-lean-cta-banner__body-content"
          >
            <p>
              It looks like your visitor groups aren’t available anymore. Learn more about how to group site visitors in 
              <a
                aria-label="Analytics (opens in a new tab)"
                class="googlesitekit-cta-link googlesitekit-cta-link--secondary"
                href="https://support.google.com/analytics/answer/12799087?hl=en-US"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Analytics
                </span>
              </a>
              .
            </p>
            <p>
              You can deactivate this widget in 
              <button
                class="googlesitekit-cta-link googlesitekit-cta-link--secondary"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Settings
                </span>
              </button>
              .
            </p>
          </div>
        </div>
        <div
          class="googlesitekit-lean-cta-banner__graphic"
        >
          <svg />
        </div>
      </div>
    </div>
  </div>
</div>
`;
